geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingCommandDUnitTest.java

```diff
@@ -834,6 +889,34 @@ public void createMappingsWithExistingPdxName() {
       assertValidEmployeeMappingOnLocator(regionMapping, region1Name, null, false, false,
           "employeeRegion");
     });
+
+    csb = new CommandStringBuilder(DESCRIBE_MAPPING).addOption(REGION_NAME,
+        region1Name);
+    CommandResultAssert commandResultAssert = gfsh.executeAndAssertThat(csb.toString());
+    commandResultAssert.containsOutput("age       | INT      | AGE         | INTEGER   | true");
+  }
+
+  @Test
+  public void createMappingsDoesNotRequirePdxSerializable() {
+    String region1Name = "region1";
+    setupReplicate(region1Name);
+
+    CommandStringBuilder csb = new CommandStringBuilder(CREATE_MAPPING);
+    csb.addOption(REGION_NAME, region1Name);
+    csb.addOption(DATA_SOURCE_NAME, "connection");
+    csb.addOption(TABLE_NAME, "employeeNumeric");
+    csb.addOption(PDX_NAME, EmployeeNumeric.class.getName());
```

Comment:
try changing 'EmployeeNumeric.class.getName()' to: "org.apache.geode.connectors.jdbc.internal.cli.CreateMappingCommandDUnitTest.EmployeeNumeric"
Does that cause it to work without your change of \\$ to \\.?

Reply:
This does not work, but this.reflectionBasedAutoSerializerFactory.create("\\Q" + className + "\\E"); will work.